Gilead-BioStats/gsm
# Add `rbm-viz` v2.1.0.

## Overview
Adds new `riskSignalSelected` event that can be monitored by any ancestor of the widget (`barChart`, `scatterPlot`, `timeSeries`, `groupOverview`).  Returns the {gsm} data associated with the clicked risk signal.

The event also fires when the widget-level group dropdown changes.  For the time series, the event should return the latest data point.

## Test Notes/Sample Code
1. Generate a report.
```
load_all()
lCharts <- MakeCharts(
  dfResults = reportingResults,
  dfGroups = reportingGroups,
  dfMetrics = reportingMetrics,
  dfBounds = reportingBounds
)

strOutputFile <- "StandardSiteReport.html"
kri_report_path <- Report_KRI(
  lCharts = lCharts,
  dfResults = reportingResults,
  dfGroups = reportingGroups,
  dfMetrics = reportingMetrics,
  strOutputFile = strOutputFile
)
browseURL('StandardSiteReport.html')
```
2. Open the report in the browser.
3. Open the browser console and add an event listener for `riskSignalSelected`:
```
document.body.addEventListener('riskSignalSelected', (event) => {
    console.table(event.data);
});
```
4. Click a point/bar and verify the underlying result is displayed in the browser console.
5. Update the widget-level group dropdown and verify the underlying result is displayed in the browser console.
    - For the time series, verify the latest result is displayed in the browser console.

## Connected Issues
- Closes #1855 
- Closes #1860
